pkg/logging/sftp/sftp_test.go

```diff
@@ -0,0 +1,255 @@
+	for _, testcase := range []struct {
+		name      string
+		cmd       *CreateCommand
+		want      *fastly.CreateSFTPInput
+		wantError string
+	}{
+		{
+			name: "all values set flag serviceID",
+			cmd:  createCommandOK(),
+			want: &fastly.CreateSFTPInput{
+				Service:           "123",
+				Version:           2,
+				Name:              fastly.String("log"),
+				Address:           fastly.String("127.0.0.1"),
+				Port:              fastly.Uint(80),
+				User:              fastly.String("user"),
+				Password:          fastly.String("password"),
+				PublicKey:         fastly.String(pgpPublicKey()),
+				SecretKey:         fastly.String(sshPrivateKey()),
+				SSHKnownHosts:     fastly.String(knownHosts()),
+				Path:              fastly.String("/log"),
+				Period:            fastly.Uint(3600),
+				FormatVersion:     fastly.Uint(2),
+				GzipLevel:         fastly.Uint(2),
+				Format:            fastly.String(`%h %l %u %t "%r" %>s %b`),
+				ResponseCondition: fastly.String("Prevent default logging"),
+				TimestampFormat:   fastly.String("%Y-%m-%dT%H:%M:%S.000"),
+				Placement:         fastly.String("none"),
+			},
+		},
+		{
+			name:      "error missing serviceID",
+			cmd:       createCommandMissingServiceID(),
+			want:      nil,
+			wantError: errors.ErrNoServiceID.Error(),
+		},
+	} {
+		t.Run(testcase.name, func(t *testing.T) {
+			have, err := testcase.cmd.createInput()
+			testutil.AssertErrorContains(t, err, testcase.wantError)
+			testutil.AssertEqual(t, testcase.want, have)
+		})
+	}
+}
+
+func TestUpdateCreateInput(t *testing.T) {
+	for _, testcase := range []struct {
+		name      string
+		cmd       *UpdateCommand
+		api       mock.API
+		want      *fastly.UpdateSFTPInput
+		wantError string
+	}{
+		{
+			name: "all values set flag serviceID",
+			cmd:  updateCommandOK(),
+			api:  mock.API{GetSFTPFn: getSFTPOK},
+			want: &fastly.UpdateSFTPInput{
+				Service:           "123",
+				Version:           2,
+				Name:              "logs",
+				NewName:           fastly.String("new-log"),
+				Address:           fastly.String("127.0.0.1"),
+				Port:              fastly.Uint(80),
+				User:              fastly.String("user"),
+				Password:          fastly.String("password"),
+				PublicKey:         fastly.String(pgpPublicKey()),
+				SecretKey:         fastly.String(sshPrivateKey()),
+				SSHKnownHosts:     fastly.String(knownHosts()),
+				Path:              fastly.String("/log"),
+				Period:            fastly.Uint(3600),
+				FormatVersion:     fastly.Uint(2),
+				GzipLevel:         fastly.Uint(2),
+				Format:            fastly.String(`%h %l %u %t "%r" %>s %b`),
+				ResponseCondition: fastly.String("Prevent default logging"),
+				TimestampFormat:   fastly.String("%Y-%m-%dT%H:%M:%S.000"),
+				Placement:         fastly.String("none"),
+			},
+		},
+		{
+			name:      "error missing serviceID",
+			cmd:       updateCommandMissingServiceID(),
+			want:      nil,
+			wantError: errors.ErrNoServiceID.Error(),
+		},
+	} {
```

Comment:
It might be worthwhile to add an additional test case that sets only the required flags and asserts that unset, optional flags aren't leading to unexpected input (`0`, `""`.)